shell/platform/windows/game_pad_winuwp.cc

```diff
@@ -0,0 +1,228 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/windows/game_pad_winuwp.h"
+
+namespace flutter {
+
+GamepadWinUWP::GamepadWinUWP(GamepadDualAxisCallback leftstick,
+                             GamepadDualAxisCallback rightstick,
+                             GamepadSingleAxisCallback lefttrigger,
+                             GamepadSingleAxisCallback righttrigger,
+                             GamepadButtonCallback pressedcb,
+                             GamepadButtonCallback releasedcb,
+                             GamepadAddedRemovedCallback changedcb)
+    : pressed_buttons_(winrt::Windows::Gaming::Input::GamepadButtons::None),
+      left_stick_x_value_(0),
+      left_stick_y_value_(0),
+      right_stick_x_value_(0),
+      right_stick_y_value_(0),
+      left_trigger_value_(0),
+      right_trigger_value_(0) {
+  left_stick_callback_ = leftstick;
+  right_stick_callback_ = rightstick;
+
+  left_trigger_callback_ = lefttrigger;
+  right_trigger_callback_ = righttrigger;
+
+  button_pressed_callback_ = pressedcb;
+  button_released_callback_ = releasedcb;
+
+  arrival_departure_callback_ = changedcb;
+}
+
+void GamepadWinUWP::Initialize() {
+  RefreshCachedGamepads();
+
+  winrt::Windows::Gaming::Input::Gamepad::GamepadAdded(
+      {this, &GamepadWinUWP::OnGamepadAdded});
+  winrt::Windows::Gaming::Input::Gamepad::GamepadRemoved(
+      {this, &GamepadWinUWP::OnGamepadRemoved});
+
+  const std::lock_guard<std::mutex> lock(gamepad_mutex_);
+  current_game_pad_ = GetLastGamepad();
+}
+
+bool GamepadWinUWP::HasController() {
```

Comment:
Could this be `const`?

You'd need to mark the mutex as `mutable` but that's encouraged. IIRC the "M&M rule" -- mutable and mutex go together :)

Reply:
Fixed in next commit.